Upgrade to Vaadin 25.1 and fix compatibility issues

Artur- opened the pull request:
- Upgrade Vaadin to 25.1.0
- Upgrade Spring Boot to 4.0.6
- Add Hilla spring-boot-starter (opt-in in Vaadin 25)
- Migrate Jackson 2 to Jackson 3 (tools.jackson.databind)
- Update Hilla annotations: @Endpoint → @BrowserCallable
- Update JSpecify annotations: @Nonnull → @NonNull
- Replace deprecated Label with Span component
- Replace deprecated Details.setContent() with removeAll() + add()
- Migrate from ListenableFuture to CompletableFuture
- Update EventData API to use JsonNode methods (asText, asDouble)
- Simplify JSON parsing in GridExcelLikeCellSelection using JsonNode
- Update Grid API: setClassNameGenerator → setPartNameGenerator
- Update CSS selectors from class-based to part-based ([part~="name"])
- Add ErrorDetails record for structured browser error logging
- Use double[] array for map coordinates instead of separate parameters
- Update frontend dependencies (Vite, TypeScript, React plugins)

Artur- committed:
Upgrade to Vaadin 25 and fix compatibility issues

- Upgrade Vaadin from 24.9.5 to 25.0.0-beta6
- Upgrade Spring Boot from 3.5.8 to 4.0.0
- Add Hilla spring-boot-starter (opt-in in Vaadin 25)
- Migrate Jackson 2 to Jackson 3 (tools.jackson.databind)
- Update Hilla annotations: @Endpoint → @BrowserCallable
- Update JSpecify annotations: @Nonnull → @NonNull
- Replace deprecated Label with Span component
- Replace deprecated Details.setContent() with removeAll() + add()
- Migrate from ListenableFuture to CompletableFuture
- Update EventData API to use JsonNode methods (asText, asDouble)
- Simplify JSON parsing in GridExcelLikeCellSelection using JsonNode
- Update Grid API: setClassNameGenerator → setPartNameGenerator
- Update CSS selectors from class-based to part-based ([part~="name"])
- Add ErrorDetails record for structured browser error logging
- Use double[] array for map coordinates instead of separate parameters
- Update frontend dependencies (Vite, TypeScript, React plugins)

manolo commented:
Running the app with `mvn` fails with 

```
2025-11-24T08:49:41.928+01:00 ERROR 23907 --- [  restartedMain] o.a.c.c.C.[Tomcat].[localhost].[/]       : Exception sending context initialized event to listener instance of class [com.vaadin.flow.spring.VaadinServletContextInitializer$CompositeServletContextListener]

java.lang.NoClassDefFoundError: com/vaadin/flow/component/textfield/HasPrefixAndSuffix
```

Artur- committed:
vaadin 25.0.3, spring boot 4.0.1

Artur- committed:
Fix compilation errors for Vaadin 25

Replace javax.annotation.Nullable with org.jspecify.annotations.Nullable
to fix compilation errors. The javax.annotation package is no longer
available in Jakarta EE used by Spring Boot 4.

Also update Hilla packages from beta5 to stable 25.0.3 release.

Artur- committed:
Add vaadin-dev dependency and update viritin to 3.1.2

- Add vaadin-dev dependency required for Vaadin 25 development mode
- Update viritin from 2.21.1 to 3.1.2 for Vaadin 25 compatibility
  (fixes NoClassDefFoundError: HasPrefixAndSuffix)

Artur- committed:
Fix frontend build issues for Vaadin 25

- Add @polymer/polymer dependency for legacy component compatibility
- Remove obsolete @JsModule badge import from LargeUploadArea
- Update package-lock.json with new dependencies

Artur- committed:
Enable component styles and add glob again

Artur- committed:
Upgrade to Vaadin 25.1.0 and Spring Boot 4.0.5

Artur- committed:
cleanup

Artur- committed:
Merge remote-tracking branch 'origin/main' into vaadin-25

Artur- committed:
Fix production build by running parseClientRoutes with Node's native TS support

Vaadin 25 uses Node 24 which silently breaks tsc (exits 0, emits nothing).
Run parseClientRoutes.ts directly with --experimental-strip-types instead,
and change the RecipeInfo import to type-only so it's stripped at runtime.

Artur- committed:
Merge branch 'main' into vaadin-25
